microsoft/testsuites/core/storage.py

```diff
@@ -242,6 +245,155 @@ def verify_os_partition_identifier(self, log: Logger, node: RemoteNode) -> None:
+        max_data_disk_count = disk.get_max_data_disk_count()
+        log.debug(f"max_data_disk_count: {max_data_disk_count}")
+
+        # get the number of data disks already added to the vm
+        assert node.capability.disk
+        assert isinstance(node.capability.disk.data_disk_count, int)
+        current_data_disk_count = node.capability.disk.data_disk_count
+        log.debug(f"current_data_disk_count: {current_data_disk_count}")
+
+        # disks to be added to the vm
+        disks_to_add = max_data_disk_count - current_data_disk_count
+
+        # get partition info before adding data disk
+        partitions_before_adding_disk = node.tools[Lsblk].get_partitions(force_run=True)
+
+        for i in range(disks_to_add):
+            disk_id = self._get_managed_disk_id(str(i))
+
+            # add data disk
+            log.debug(f"Adding managed disk: {disk_id}")
+            disk.add_managed_data_disk(
+                [disk_id], [str(i + current_data_disk_count)], size=str(disk_size_in_gb)
+            )
+
+            # verify that partition count is increased by 1
+            # and the size of partition is correct
+            partitons_after_adding_disk = node.tools[Lsblk].get_partitions(
+                force_run=True
+            )
+            added_partitions = [
+                item
+                for item in partitons_after_adding_disk
+                if item not in partitions_before_adding_disk
+            ]
+            log.debug(f"added_partitions: {added_partitions}")
+            assert_that(added_partitions).is_length(1)
+            assert_that(added_partitions[0].size).is_equal_to(
+                disk_size_in_gb * 1024 * 1024 * 1024
+            )
+
+            # remove data disk
+            log.debug(f"Removing managed disk: {disk_id}")
+            disk.remove_managed_data_disk([disk_id])
+
+            # verify that partition count is decreased by 1
+            partition_after_removing_disk = node.tools[Lsblk].get_partitions(
+                force_run=True
+            )
+            added_partitions = [
+                item
+                for item in partitions_before_adding_disk
+                if item not in partition_after_removing_disk
+            ]
+            assert_that(added_partitions).is_length(0)
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the disks can be added in parallel while the
+        vm is running.
+        Steps:
+        1. Get maximum number of data disk for the current vm_size.
+        2. Get the number of data disks already added to the vm.
+        3. Add maximum number of data disks to the VM in parallel.
+        4. Verify that the disks are added are available in the OS.
+        5. Remove the disks from the vm in parallel.
+        6. Verify that the disks are removed from the OS.
+        """,
+        requirement=simple_requirement(supported_platform_type=[AZURE]),
+    )
```

Comment:
Should we implement max_disk_count similar with max_nic_count?
Then have separate test cases which can test the VM sku with max number of data disk currently on azure, 64.

Reply:
The number of data disks is variable for each vm_size. For example, `Standard_DS2_v2` can only have 8 max data disks. Please feel free to clarify if I misunderstood your comment.

Reply:
Lili is trying to handle the same problem. If the test case will create 9 data disks on Standard_DS2_v2, it will failed. But we can use the max_data_disk_count to skip this vm size, if a test case declares it needs 9 data disks. Refer to the code of max_nic_count.

Reply:
We also need to consider the test coverage; current design may only cover the minimal data disk scenario.